jdaviz/core/region_translators.py

```
@@ -504,6 +504,43 @@ def stcs_string2region(stcs_string):
     return SKY_REGION_FROM_COORDS_FACTORY[shape](coordinates, frame=frame)
 
 
+def region2stcs_string(region):
+    """Convert a `regions.Region` object to an STC-S string.
+
+    Currently supports only `CircleSkyRegion` and `EllipseSkyRegion`.
+
+    Parameters
+    ----------
+    region : `regions.Region`
+
+    Returns
+    -------
+    stcs_str : str
+        STC-S string representing the region, using the format:
+        - `CIRCLE <FRAME> <RA> <DEC> <RADIUS>`
+        - `ELLIPSE <FRAME> <RA> <DEC> <WIDTH> <HEIGHT> <ANGLE>`
+    """
+    if isinstance(region, CircleSkyRegion):
+        shape = "CIRCLE"
+        coords = [region.center.ra.deg, region.center.dec.deg, region.radius.to_value(u.deg)]
+    elif isinstance(region, EllipseSkyRegion):
+        shape = "ELLIPSE"
+        coords = [region.center.ra.deg, region.center.dec.deg,
+                  region.width.to_value(u.deg),
+                  region.height.to_value(u.deg),
+                  region.angle.to_value(u.deg)]
+    else:
+        raise NotImplementedError(f"{type(region).__name__} is not supported for STC-S export")
+
+    frame = region.center.frame.name.upper()
+    if frame not in SUPPORTED_STCS_FRAME_VALUES:
+        frame = "ICRS"

```

Question: is `"ICRS"` the default value per the specification? I would expect that we would want to raise an exception here if the frame value wasn't supported by STCS

---

You're absolutely right. I was silently falling back to `ICRS` as a default, but I can see how that could propagate misleading metadata. I appreciate your point, and I agree that raising an error would be a better approach here. I'll update the code to raise a ValueError if the frame isn't one of the supported STC-S values, and include a note listing the valid options.

---

I agree that the API call should show an error. We should also ensure that users can select regions for export as STC-S in the Export plugin dropdown menu _only_ if that region is exportable.